### ISSUE TITLE
swarm cluster shows incorrect or improper node number

### ISSUE THREAD

**Issue by allencloud** [docker/docker#30179]
**Description**

Hi, All, 

I found that in a swarm cluster, when I use `docker swarm leave -f` on a worker and at once use `docker swarm join`, the `docker node ls` will show incorrect node number and improper node number.

I have only two nodes: A and B.

**Steps to reproduce the issue:**
1. docker swarm init (on a manager A)
2. docker swarm join (on a worker B)
3. docker swarm leave -f (on worker B)
4. docker swarm join ( on worker B)
5. docker node ls (on manager A)

**Describe the results you received:**
After step 4, I got 
```
root@ubuntu:~# docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
34sr1zxx43djkr878c8sltcby    ubuntu    Ready   Active
gu2skiscf5ddgh7wzd2u9ih75    ubuntu    Ready   Active
ulbs34n7h46d7th4nbrq03ohs *  ubuntu    Ready   Active        Leader
```

After step 5, I got 
```
root@ubuntu:~# docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
34sr1zxx43djkr878c8sltcby    ubuntu    Down    Active
avzbj8l22g0rfl4ei2b1tm5zq    ubuntu    Ready   Active
gu2skiscf5ddgh7wzd2u9ih75    ubuntu    Ready   Active
ulbs34n7h46d7th4nbrq03ohs *  ubuntu    Ready   Active        Leader
```

After a few seconds of step 5, I got 
```
root@ubuntu:~# docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
34sr1zxx43djkr878c8sltcby    ubuntu    Down    Active
avzbj8l22g0rfl4ei2b1tm5zq    ubuntu    Ready   Active
gu2skiscf5ddgh7wzd2u9ih75    ubuntu    Down    Active
ulbs34n7h46d7th4nbrq03ohs *  ubuntu    Ready   Active        Leader
```

**Describe the results you expected:**
 Since I only have two machines, when I made B(single worker) leave and made it join again, swarm can be conscious the reconnect of worker B.  Then the node added will overwrite the `Down` one.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
root@ubuntu:~# docker version
Client:
 Version:      1.13.0-rc4
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   88862e7
 Built:        Fri Dec 16 22:59:15 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-rc4
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   88862e7
 Built:        Fri Dec 16 22:59:15 2016
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
root@ubuntu:~# docker info
Containers: 8
 Running: 0
 Paused: 0
 Stopped: 8
Images: 30
Server Version: 1.13.0-rc4
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 41
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: ulbs34n7h46d7th4nbrq03ohs
 Is Manager: true
 ClusterID: bbq18zj8f816d1iewpwsbc4jh
 Managers: 1
 Nodes: 4
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.59.104
 Manager Addresses:
  192.168.59.104:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 51371867a01c467f08af739783b8beafc154c4d7
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.19.0-25-generic
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.954 GiB
Name: ubuntu
ID: FXOY:JCOY:HKDI:VO5U:FYDM:UEXV:YIFN:AISM:NR6U:VMW5:V4MZ:RQWF
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 32
 Goroutines: 141
 System Time: 2017-01-16T15:21:27.456686468+08:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by allencloud** [comment]
I am also wondering if this is correct:
```
root@ubuntu:~# docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
34sr1zxx43djkr878c8sltcby    ubuntu    Down    Active
```
With a status of `Down`, but the availability is `Active`?

**Comment by tonistiigi** [comment]
To remove the node you have to run `docker node rm`. Leave is an operation for removing cluster data local to the current node and doesn't modify cluster state(#24088). I believe "availability" is completely independent of status and only represents how the node was configured. I can see how it may be confusing if the value is "active".

**Comment by thaJeztah** [comment]
yes, this is a duplicate of https://github.com/docker/docker/issues/24088; let's close this one so that the discussion doesn't diverge 👍
